lib/prefetch.js

```diff
@@ -116,20 +16,23 @@ function getPrefetchUrl (href) {
 }
 
 export async function prefetch (href) {
-  if (!hasServiceWorkerSupport()) return
+  if (typeof window === 'undefined') return
   if (!isLocal(href)) return
 
-  // Register the service worker if it's not.
-  messenger.ensureInitialized()
-
   const url = getPrefetchUrl(href)
-  if (!PREFETCHED_URLS[url]) {
-    PREFETCHED_URLS[url] = messenger.send({ action: 'ADD_URL', url: url })
-  }
+  if (PREFETCHED_URLS[url]) return PREFETCHED_URLS[url]
 
+  PREFETCHED_URLS[url] = fetch(url)
```

Comment:
We could try to do this after the `load` event.
When we reach this page, we already passed `DOMContentLoaded`. 

Since we don't do any processing, this should fine.